_posts/2016-07-20-Automation-testing-with-postman.md

```diff
@@ -0,0 +1,164 @@
+<br/>
+<br/>
+Scroll down and hit the blue "Start Test" button. Following test report will be presented to you.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/runner_result.jpg)
+<br/>
+<br/>
+Writing a request and tests for each different permutation of data could get tiresome and tedious. On the test runner screen we are given the option to choose a data file. This data file can be either a CSV or a JSON file, but will allow us to set up data in bulk to be run through the test runner. Create a new csv file and copy following snippet into it.
+
+~~~~
+input,expected_result
+2,"0.69314718056"
+224,"5.41164605186"
+3000,"8.00636756765"
+388949,"12.8712035086"
+~~~~
+
+We need to rewrite the body of our request so it will use the variable of our csv as follows.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/request_csv.jpg)
+<br/>
+<br/>
+We also need to rewrite our test. Like you can see we use the 'data' object to call our expected_result variable.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/test_csv.jpg)
+<br/>
+<br/>
+Back to the runner window. Select the "Calculator" collection and the "Production" environment. Click the "Choose Files" button and select the csv file you just created, click the "Preview" button to check for any inconsistenties. As there are 4 entries in our csv we want to use to feed our test enter 4 in the iteration inputfield.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/runner_csv.jpg)
+<br/>
+<br/>
+Hit the "Start Test" button and you will now see 12 green tests. Pretty neat, isn't it?
+<br/>
+<br/>
+![Alt text](/img/postman-2016/runner_result_csv.jpg)
+<br/>
+<br/>
+
+### Newman
+Integrating Postman tests with build systems can easily be accomplished with Newman. Newman is the command line tool companion for Postman. It can be installed through the Node.js package manager, npm. You'll find more information on how the install Newman [here](https://github.com/postmanlabs/newman){:target="_blank"}.
+
+After Newman is installed we can export our previously created collection and environment. Select the 'Calculator' collection and hit export and save as 'my_collection.json'.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/export_collection.jpg)
+<br/>
+<br/>
+To export the 'Production' environment select 'Manage Environment' and on the next popup hit export and save as 'prod_environment.json'.
+<br/>
+<br/>
+![Alt text](/img/postman-2016/export_environment.jpg)
+<br/>
+<br/>
+
+Now run you test with Newman using following command where my_collection.json is the exported collection, my_data.csv the csv, prod_environment.json the environment and -n the number of lines from our csv.
+
+~~~~
+newman run my_collection.json -n 4 -d my_data.csv -e prod_environment.json
+~~~~
+
+### Summary
+In this tutorial we saw how to create a request and a test. We saw how to create a collection and how to run it with the collection runnner and Newman.
+I hope you enjoyed this tutorial and if you have any question feel free.
```

Comment:
if you have any question feel free. => if you have any question feel free to ... 
(replace ... by "to add these as a comment" or "to email me at ..." or ...